src/teleop/gui/src/components/ArmControls.vue

```diff
@@ -1,99 +1,203 @@
+      ros: this.$ros,
+      name: "/joint_lock",
+      messageType: "mrover/JointLock"
+    });
+    this.jointstate_pub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/ra_cmd",
+      messageType: "sensors_msgs/JointState"
+    });
+    const jointData = {
+      //publishes array of all falses when refreshing the page
+      joints: this.joints_array
+    };
+    var jointlockMsg = new ROSLIB.Message(jointData);
+    this.jointlock_pub.publish(jointlockMsg);
+
+    const updateRate = 0.1;
     interval = window.setInterval(() => {
-      if(this.arm_enabled){
-        const gamepads = navigator.getGamepads()
+      if (this.arm_enabled) {
+        const gamepads = navigator.getGamepads();
         for (let i = 0; i < 4; i++) {
-          const gamepad = gamepads[i]
+          const gamepad = gamepads[i];
           if (gamepad) {
-            if (gamepad.id.includes('Microsoft') || gamepad.id.includes('Xbox')) {
-              let buttons = gamepad.buttons.map((button) =>{
-                  return button.value
-                })
-  
-  
-              const joystickData = {
-                  axes: gamepad.axes,
-                  buttons: buttons
-              }
-              var joystickMsg = new ROSLIB.Message(joystickData)
-              this.joystick_pub.publish(joystickMsg)
+            if (
+              gamepad.id.includes("Microsoft") ||
+              gamepad.id.includes("Xbox")
+            ) {
+              let buttons = gamepad.buttons.map((button) => {
+                return button.value;
+              });
+              this.publishJoystickMessage();
             }
           }
         }
       }
-    }, updateRate*1000)
+    }, updateRate * 1000);
   },
 
-    methods: {
-        updateArmEnabled: function (enabled){
-            this.arm_enabled = enabled
+  methods: {
+    updateArmEnabled: function (enabled) {
+        this.arm_enabled = enabled;
+        //ra_command, have all nulls in position. And all zeros in velocity array
+        if(arm_enabled === false){
+            const zeroCommandData = {
+            name: ["joint_a", "joint_b", "joint_c", "joint_d", "joint_e", "joint_f"],
+            position: [math.nan, math.nan, math.nan, math.nan, math.nan, math.nan],
+            velocity: [0, 0, 0, 0, 0, 0],
+            effort: [0.5, 0.5, 0.5, 0.5, 0.5, 0.5],
+            }
+            var zeroCommandMsg = new ROSLIB.Message(zeroCommandData);
+            this.jointstate_pub.publish(zeroCommandMsg);
         }
     },
-
-    components: {
-        Checkbox
+    updateServo: function (enabled) {
+      this.servo = enabled;
+    },
+    updateJointsEnabled: function (jointnum, enabled) {
+      this.joints_array[jointnum] = enabled;
+      const jointData = {
+        joints: this.joints_array
+      };
+      var jointlockMsg = new ROSLIB.Message(jointData);
+      this.jointlock_pub.publish(jointlockMsg);
+    },
+    publishJoystickMessage: function(){
+        const joystickData = {
```

Comment:
You'll need to actually pass the axis and button values to this for it to actually work.